app/main.py

```
@@ -1,34 +1,89 @@
+        for desk in self.decks:
+            if desk.row == row and desk.column == column:
+                return desk
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> str:
+        fire_desk = self.get_deck(row, column)
+        fire_desk.is_alive = False
+        for desk in self.decks:
+            if desk.is_alive:
+                return "Hit!"
+        self.is_drowned = True
+        return "Sunk!"
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def check_neighbor_cells(self, ship: Ship, ship_desk: tuple) -> None:
+        for row in range(ship_desk[0] - 1, ship_desk[0] + 2, 2):
+            for column in range(ship_desk[1] - 1, ship_desk[1] + 2, 2):
+                if ((row, column) in self.field) \
+                        and (self.field[(row, column)] != ship):
+                    print("ships shouldn't be located "
+                          "in the neighboring cells")
+
+    def _validate_field(self) -> None:
+        ships = set([ship for ship in self.field.values()])
+        deck_ships = {4: 0, 3: 0, 2: 0, 1: 0}
+        for ship in ships:
+            deck_ships[len(ship.decks)] += 1
+            self.check_neighbor_cells(ship, ship.start)
+            self.check_neighbor_cells(ship, ship.end)
+        if deck_ships != {4: 1, 3: 2, 2: 3, 1: 4}:
+            print("the total number of the ships should be 10:\n"
```

use raise exception instead of print